opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1254,6 +1275,17 @@ namespace {
         struct Chop01 {
             double operator()(double x) const { return std::max(std::min(x, 1.0), 0.0); }
         };
+
+        double infinityNorm( const ADB& a )
+        {
+            if( a.value().size() > 0 ) {
+                return a.value().matrix().template lpNorm<Eigen::Infinity> ();
```

Comment:
This breaks the build on GCC 4.4 and GCC 4.6 (although not on GCC 4.8).  You can't use the `.template` construction outside templates.  Specifically, neither `a` nor its `value().matrix()` depend on a template parameter so the use of `.template` is neither required nor permitted.

Of course, this was wrong in the initial version too, the one you refactored, but I didn't pick up on it until now.